jaeger-spark-dependencies-common/src/main/java/io/jaegertracing/spark/dependencies/DepenencyLinksSparkJob.java

```diff
@@ -0,0 +1,43 @@
+/**
+ * Copyright 2017 The Jaeger Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package io.jaegertracing.spark.dependencies;
+
+import io.jaegertracing.spark.dependencies.model.Dependency;
+import io.jaegertracing.spark.dependencies.model.Span;
+import java.util.List;
+import org.apache.spark.api.java.JavaPairRDD;
+import scala.Tuple2;
+
+/**
+ * @author Pavol Loffay
+ */
+public class DepenencyLinksSparkJob {
+
+  private DepenencyLinksSparkJob() {}
+
+  /**
+   * Derives dependency links based on supplied spans.
```

Comment:
Can you expand this javadoc, with more information about the input and output, perhaps with examples? One question: if all I have are single spans (ie: traces of 1 span), would I ever get one iteration on this method? Or would this method process only spans related to the same trace ID?

Reply:
>  One question: if all I have are single spans (ie: traces of 1 span), would I ever get one iteration on this method? Or would this method process only spans related to the same trace ID?

This is more complicated as it is RDD.  `traceIdSpans.flatMapValues` method is called per trace.